api/provider/aws/environment_delete.go

```diff
@@ -53,6 +59,42 @@ func (e *EnvironmentProvider) Delete(environmentID string) error {
 	return nil
 }
 
+func (e *EnvironmentProvider) checkEnvironmentDependencies(environmentID string) error {
+	deleteEntity := func(entity string) error {
+		tags, err := e.TagStore.SelectByType(entity)
+		if err != nil {
+			return err
+		}
+
+		dependentTags := tags.WithKey("environment_id").WithValue(environmentID)
+		if len(dependentTags) > 0 {
+			IDs := []string{}
+			for _, tag := range dependentTags {
+				IDs = append(IDs, tag.EntityID)
+			}
+
+			msg := "Cannot delete environment '%s' because it contains dependent %s: %s"
+			msg = fmt.Sprintf(msg, environmentID, entity, strings.Join(IDs, ", "))
```

Comment:
[nit] this is a little different then how we typically do long strings. I think it's a littler clearer what's going on by doing `fmt.Sprintf` in-line as opposed to having that in separate lines. For example:
```
msg := fmt.Sprintf("Cannot delete environment while it contains dependent %ss: ", entityType)
msg += strings.Join(entityIDs, ", ")
return errors.New(...)
```

Reply:
Yeah I agree with this. It makes it easier to read.